broke-broker/src/main/java/org/dsngroup/broke/broker/storage/SubscriberPool.java

```diff
@@ -0,0 +1,83 @@
+package org.dsngroup.broke.broker.storage;
+
+import io.netty.channel.ChannelHandlerContext;
+import org.dsngroup.broke.broker.SubscriberContext;
+import org.dsngroup.broke.protocol.PublishMessage;
+
+import java.util.*;
+
+/**
+ * Subscriber pool class that stores a map.
+ * The map maps the topic to its subscriber set.
+ * This can be a singleton.
+ * */
+public class SubscriberPool {
+
+    /**
+     * The subscriber pool
+     * A map with topic key to a map of subscribers
+     * The map of subscribers maps the ChannelHandlerContext instance to a subscriber context.
+     * (Because the ChannelHandlerContext of each subscriber is unique).
+     * */
+    private static Map<String, Map<ChannelHandlerContext, SubscriberContext>> subscriberPool = new HashMap();
+
+    /**
+     * Register the subscriber to an interested topic
+     * @param topic Interested topic
+     * @param groupId consumer group id
+     * @param channelCxt subscriber's ChannelHandlerContext {@see ChannelHandlerContext}
+     * @return the subscriber's instance
+     * */
+    public static SubscriberContext register(String topic, String groupId, ChannelHandlerContext channelCxt) throws Exception{
+
+        if (subscriberPool.get(topic)==null)
+            subscriberPool.put(topic, new TreeMap<>( new ChannelHandlerComparator()));
+
+        SubscriberContext subscriber = new SubscriberContext(topic, groupId, channelCxt);
+
+        subscriberPool.get(topic).put(channelCxt, subscriber);
+
+        // TODO: remove this test
+        // System.out.println( "Test the subscriber pool: \n"+subscriberPool.get(topic).get(channelCxt).getClass().toString() );
+        return subscriber;
+    }
+
```

Comment:
I think we can also consider how to generate a more well-defined data structure / classes between `SubscriberPool` and `InMemoryPool`, in the future.